src/services/rfqm_service.ts

```diff
@@ -490,20 +514,25 @@ export class RfqmService {
             gasPrice = await this._protocolFeeUtils.getGasPriceEstimationOrThrowAsync();
         } catch (error) {
             logger.error({ error }, 'Current gas price is unable to be fetched, marking worker as not ready.');
+            RFQM_WORKER_NOT_READY.labels(workerAddress).inc();
             return false;
         }
 
+        const balance = await this._blockchainUtils.getAccountBalanceAsync(workerAddress);
+        const balanceUnitAmount = Web3Wrapper.toUnitAmount(balance, ETH_DECIMALS).decimalPlaces(PRICE_DECIMAL_PLACES);
+        RFQM_WORKER_BALANCE.labels(workerAddress).set(balanceUnitAmount.toNumber());
+
         // check for outstanding jobs from the worker and resolve them
         const unresolvedJobs = await this._dbUtils.findUnresolvedJobsAsync(workerAddress);
+        RFQM_JOB_REPAIR.labels(workerAddress).inc(unresolvedJobs.length);
         for (const job of unresolvedJobs) {
             logger.info({ workerAddress, orderHash: job.orderHash }, `Unresolved job found, attempting to re-process`);
             await this.processRfqmJobAsync(job.orderHash, workerAddress);
         }
 
-        const balance = await this._blockchainUtils.getAccountBalanceAsync(workerAddress);
         const isWorkerReady = await this._blockchainUtils.isWorkerReadyAsync(workerAddress, balance, gasPrice);
```

Comment:
i think somewhere down this rabbit hole of logic the unresolved jobs check gets run again.. we should refactor at some point

Reply:
Yes, this happens here: https://github.com/0xProject/0x-api/pull/670/files#diff-289e53810d2a508fae0fff19710e2e2b9846a98a8935466d7e20a1490f179376L498